### PR TITLE
[github-actions] use cmake to build nrf in size action

### PR DESCRIPTION
**Pull request by Irving-cl**
This PR updates nrf building in size check to use cmake with the latest cmake support. The PR also refines the shell implementation.

### CODE REVIEW

**Comment by Irving-cl** [comment]
This PR depends on #6127 and cannot be merged until #6127 get merged.

I updated the output directory of binaries and libraries for cmake nrf build and referenced the new directory in this PR when doing size check. However the size action pulls the old version openthread (which doesn't update the output directory). This would cause size action fail.

**Comment by jwhui** [comment]
@Irving-cl , we are in the process of migrating example platforms into separate repos. How about we update the script to build using https://github.com/openthread/ot-nrf528xx instead?

**Comment by Irving-cl** [comment]
@jwhui 

Sure.

**Comment by Irving-cl** [comment]
Depends on [ot-nrf528xx/#3](https://github.com/openthread/ot-nrf528xx/pull/3).

File: script/check-size [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 #
-#  Copyright (c) 2019, The OpenThread Authors.
+#  Copyright (c) 2021, The OpenThread Authors.
```
Comment (simonlingoogle) [inline script/check-size:3]:
```suggestion
#  Copyright (c) 2019, The OpenThread Authors.
```

why change the year?

File: script/check-size [hunk 2]
```diff
@@ -100,101 +106,116 @@ markdown()
...
+        "-DOT_CHANNEL_MONITOR=ON"
+        "-DOT_CHILD_SUPERVISION=ON"
+        "-DOT_COAP=ON"
+        "-DOT_COAPS=ON"
+        "-DOT_COMMISSIONER=ON"
+        "-DOT_DATASET_UPDATER=ON"
+        "-DOT_DHCP6_CLIENT=ON"
+        "-DOT_DHCP6_SERVER=ON"
+        "-DOT_DIAGNOSTIC=ON"
+        "-DOT_DNS_CLIENT=ON"
```
Comment (simonlingoogle) [inline script/check-size:122]: add OT_DNSSD_SERVER=ON ?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1: 

with nits

**Comment by Irving-cl** [comment]
@jwhui 

This PR is also affected by #6133. Shall we merge it?

**Review by jwhui — APPROVED** [review]

**Pull request merged**